### PR TITLE
Vehicle ai tier2

### PR DESCRIPTION
**Pull request by LiamMillerSE**

### CODE REVIEW

File: Assets/Scripts/WorldObjects/WeaponDrop/WeaponDrop.cs [hunk 1]
```diff
@@ -80,7 +80,8 @@ private void SetGunType(PlayerWeaponType gunType)
     /// <returns>The selected PlayerGunType</returns>
     public PlayerWeaponType SetRandomGunType()
     {
-        PlayerWeaponType typeToAssign = (PlayerWeaponType) Random.Range((int)PlayerWeaponType.DefaultGun, (int)PlayerWeaponType.INVALID - 1);
+        //PlayerWeaponType typeToAssign = (PlayerWeaponType) Random.Range((int)PlayerWeaponType.DefaultGun, (int)PlayerWeaponType.INVALID - 1);
```
Comment (toryjleo) [inline Assets/Scripts/WorldObjects/WeaponDrop/WeaponDrop.cs:83]: Do we still use this? Please delete this line if we are not using this.

File: Assets/Scripts/Enemies/VehicleAI.cs [hunk 2]
```diff
@@ -39,8 +105,20 @@ void OnCollisionEnter(Collision collision)
...
         }
     }
+
+    public void CarDeath()
+    {
+        float minMaxTorque = 1800f;
+        rb.angularDrag = 1;
+        rb.constraints = RigidbodyConstraints.None;
+        rb.AddForce(new Vector3(0, 20f, 0), ForceMode.Impulse);
+        rb.AddTorque(new Vector3(Random.Range(-minMaxTorque, minMaxTorque), Random.Range(-minMaxTorque, minMaxTorque), Random.Range(-minMaxTorque, minMaxTorque)), ForceMode.Impulse);
```
Comment (toryjleo) [inline Assets/Scripts/Enemies/VehicleAI.cs:120]: Col 120 limit

File: Assets/Scripts/Enemies/VehicleAI.cs [hunk 3]
```diff
@@ -39,8 +105,20 @@ void OnCollisionEnter(Collision collision)
             Rigidbody bikeRB = bike.movementComponent.rb;
             Debug.DrawLine(transform.position, transform.position + vehicleController.carVelocity);
             bike.movementComponent.TakeDamage(DamageMultiplyer * Mathf.Abs(bike.movementComponent.appliedForce.magnitude - vehicleController.carVelocity.magnitude));
-            bikeRB.AddForce((vehicleController.carVelocity - bike.movementComponent.appliedForce) * RamModifier, ForceMode.Impulse);
+            Vector3 bumpForce = Vector3.ClampMagnitude((vehicleController.carVelocity - bike.movementComponent.appliedForce) * RamModifier, MAX_RAM_MAGNITUDE);
```
Comment (toryjleo) [inline Assets/Scripts/Enemies/VehicleAI.cs:108]: Col 120 limit

File: Assets/3rd_Party_Packages/Arcade Vehicle Ai/Scripts/ArcadeAiVehicleController.cs [hunk 4]
```diff
@@ -11,6 +11,7 @@ public enum MovementMode { Velocity, AngularVelocity };
     public LayerMask drivableSurface;
 
     public float MaxSpeed, accelaration, turn;
+    //rb is the sphere collidaer that acts as the center of balance, and the bottom bound of the vehicle
```
Comment (toryjleo) [inline Assets/3rd_Party_Packages/Arcade Vehicle Ai/Scripts/ArcadeAiVehicleController.cs:14]:
```suggestion
    //rb is the sphere collider that acts as the center of balance, and the bottom bound of the vehicle
```

**Review by toryjleo — CHANGES_REQUESTED** [review]

File: Assets/Scripts/WorldObjects/WeaponDrop/WeaponDrop.cs (hunk 1 above)
Comment (LiamMillerSE) [reply Assets/Scripts/WorldObjects/WeaponDrop/WeaponDrop.cs:83]: Resolved

File: Assets/Scripts/Enemies/VehicleAI.cs (hunk 2 above)
Comment (LiamMillerSE) [reply Assets/Scripts/Enemies/VehicleAI.cs:120]: Newlined

File: Assets/Scripts/Enemies/VehicleAI.cs (hunk 3 above)
Comment (LiamMillerSE) [reply Assets/Scripts/Enemies/VehicleAI.cs:108]: Newlined and commented

File: Assets/3rd_Party_Packages/Arcade Vehicle Ai/Scripts/ArcadeAiVehicleController.cs (hunk 4 above)
Comment (LiamMillerSE) [reply Assets/3rd_Party_Packages/Arcade Vehicle Ai/Scripts/ArcadeAiVehicleController.cs:14]: Removed a

**Review by toryjleo — APPROVED** [review]

**Pull request merged**